nemo/collections/nlp/models/question_answering/qa_model.py

```diff
@@ -184,29 +184,42 @@ def inference(
         self,
         file: str,
         batch_size: int = 1,
+        num_samples: int = -1,
         output_nbest_file: Optional[str] = None,
         output_prediction_file: Optional[str] = None,
-    ) -> List[int]:
+    ):
         """
-        Get prediction for the queries
+        Get prediction for unlabeled inference data
         Args:
-            queries: text sequences
+            file: inference data
             batch_size: batch size to use during inference
-            max_seq_length: sequences longer than max_seq_length will get truncated. default -1 disables truncation.
+            num_samples: number of samples to use of inference data. Default: -1 if all data should be used.
+            output_nbest_file: optional output file for writing out nbest list
+            output_prediction_file: optional output file for writing out predictions
         Returns:
-            all_preds: model predictions
+            all_predictions: model predictions
+            all_nbest: model nbest list
         """
         # store predictions for all queries in a single list
         all_predictions = []
         all_nbest = []
         mode = self.training
-        device = next(self.parameters()).device
+        device = 'cuda' if torch.cuda.is_available() else 'cpu'
```

Comment:
What would happen if we have multiple GPUs?

Reply:
Its still cuda

Reply:
shouldn't that take into consideration `trainer.gpus` as well?

Reply:
@tkornuta-nvidia  how to you access that using the model?

Reply:
yeah, in `inference()`/`inference with ONNX()` we will have to deal with that slightly different - as in both cases there will be no trainer object. 

But still the entry point script will have to pass `gpus` from TLT launcher - so I guess I would simply pass that argument to inference() method - we will deal with that later.